lib/mutant/ast/regexp/transformer/recursive.rb

```diff
@@ -0,0 +1,51 @@
+module Mutant
+  module AST
+    module Regexp
+      class Transformer
+        class Recursive < self
+          %i[
+            regexp_alternation_meta
+            regexp_nlookahead_assertion
+            regexp_passive_group
+            regexp_nlookbehind_assertion
+            regexp_lookbehind_assertion
+            regexp_lookahead_assertion
+            regexp_atomic_group
+            regexp_capture_group
+          ].map(&method(:register))
+
+          class ExpressionToAST < Transformer::ExpressionToAST
+            def call
+              quantify(ast(*children))
+            end
+          end
+
+          class ASTToExpression < Transformer::ASTToExpression
+            include LookupTable
+
+            # rubocop:disable LineLength
+            TABLE = IceNine.deep_freeze(
+              regexp_alternation_meta:      [::Regexp::Token.new(:meta,      :alternation, '|'),    ::Regexp::Expression::Alternation],
+              regexp_nlookahead_assertion:  [::Regexp::Token.new(:assertion, :nlookahead,  '(?!'),  ::Regexp::Expression::Assertion::NegativeLookahead],
+              regexp_passive_group:         [::Regexp::Token.new(:group,     :passive,     '(?:'),  ::Regexp::Expression::Group::Passive],
+              regexp_nlookbehind_assertion: [::Regexp::Token.new(:assertion, :nlookbehind, '(?<!'), ::Regexp::Expression::Assertion::NegativeLookbehind],
+              regexp_lookbehind_assertion:  [::Regexp::Token.new(:assertion, :lookbehind,  '(?<='), ::Regexp::Expression::Assertion::Lookbehind],
+              regexp_lookahead_assertion:   [::Regexp::Token.new(:assertion, :lookahead,   '(?='),  ::Regexp::Expression::Assertion::Lookahead],
+              regexp_atomic_group:          [::Regexp::Token.new(:group,     :atomic,      '(?>'),  ::Regexp::Expression::Group::Atomic],
+              regexp_capture_group:         [::Regexp::Token.new(:group,     :capture,     '('),    ::Regexp::Expression::Group::Capture]
+            )
+            # rubocop:enable LineLength
+
+            def call
+              expression = expression_class.new(expression_token)
```

Comment:
Instance of https://github.com/mbj/mutant/pull/565/commits/b3b6c22e8868baee9691c75127e3baba833ce8c7#r61765189